testsuite/cluster-test/src/experiments/versioning_test.rs

```diff
@@ -200,25 +202,24 @@ impl Experiment for ValidatorVersioning {
             .load_diem_root_account(&new_client)
             .await?;
         let allowed_nonce = 0;
-        let update_txn = create_user_txn(
-            &diem_root_account.key_pair,
-            TransactionPayload::Script(encode_update_diem_version_script(allowed_nonce, 11)),
-            diem_root_account.address,
-            diem_root_account.sequence_number,
-            123456,
-            0,
-            XUS_NAME.to_owned(),
-            10,
-            ChainId::test(),
-        )
-        .map_err(|e| format_err!("Failed to create signed transaction: {}", e))?;
-        diem_root_account.sequence_number += 1;
+        let update_txn = diem_root_account.sign_with_transaction_builder(
+            TransactionFactory::new(ChainId::test())
+                .payload(TransactionPayload::Script(
+                    encode_update_diem_version_script(allowed_nonce, 11),
+                ))
+                .sender(diem_root_account.address())
+                .sequence_number(diem_root_account.sequence_number())
+                .max_gas_amount(123456)
+                .gas_unit_price(0)
+                .gas_currency_code(XUS_NAME)
+                .expiration_timestamp_secs(10),
+        );
         execute_and_wait_transactions(&mut new_client, &mut diem_root_account, vec![update_txn])
             .await?;
 
         info!("7. Send the transaction using the new feature after Diem version update");
-        let txn5 = txn_gen(&account)?;
-        account.sequence_number += 1;
+        let txn5 = txn_gen(&mut account);
+        *account.sequence_number_mut() += 1;
```

Comment:
you shouldn't need to bump the seq number anymore, that should be handled internally, this might actually break the tests.